src/services/shims.ts

```diff
@@ -322,6 +322,22 @@ module ts {
         }
     }
 
+    var globalLogger: Logger;
+
+    function serialize(target: Function, key: string| symbol, descriptor: PropertyDescriptor) {
+        let action = descriptor.value;
+        descriptor.value = function () {
+            var _this = this;
+            var _arguments = arguments;
+            debugger;
+            var result = forwardJSONCall(globalLogger,
+                <string>key + "(" + Array.prototype.join.call(_arguments, ", ") + ")",
+                () => action.apply(_this, _arguments));

```

Comment:
why do you need _this.  Won't the () =>  compilation automatically handle that for you?